tests/_helpers/setup.ts

```diff
@@ -105,3 +103,25 @@ export function setup<Elements extends Element | Element[] = HTMLElement>(
     ...render<Elements>(ui, {eventHandlers, focus, selection}),
   }
 }
+export function getFocusableElement(
+  parent: Element | ShadowRoot,
+): HTMLElement | null {
+  const possibleFocusableElement: HTMLElement | null =
+    parent.querySelector(FOCUSABLE_SELECTOR)
+  if (possibleFocusableElement) {
+    return possibleFocusableElement
+  }
+
+  const children = Array.from(parent.children)
+  for (const child of children) {
+    if ('shadowRoot' in child && child.shadowRoot) {
+      const possibleFocusableChildElement = getFocusableElement(
+        child.shadowRoot,
+      )
+      if (possibleFocusableChildElement) {
+        return possibleFocusableChildElement
+      }
+    }
+  }
+  return null
+}
```

Comment:
Could you explain your preference over:
https://github.com/testing-library/user-event/pull/1033#discussion_r947111604

Reply:
None really. I can also use that if that works better for you. This basically just comes from my tinkering with delegatesFocus.

Reply:
If there is no advantage, i'd prefer the shorter implementation that takes the first focusable regardless of whether it's inside a shadow tree and there are other focusable elements after that.

Reply:
Done